src/System.Private.Xml/src/System/Xml/XmlDownloadManagerAsync.cs

```diff
@@ -34,22 +34,22 @@ internal partial class XmlDownloadManager
         private async Task<Stream> GetNonFileStreamAsync(Uri uri, ICredentials credentials, IWebProxy proxy,
             RequestCachePolicy cachePolicy)
         {
-            HttpClient client = new HttpClient();
-            HttpRequestMessage req = new HttpRequestMessage(HttpMethod.Get, uri);
-
-            lock (this)
+            WebRequest req = WebRequest.Create(uri);
+            if (credentials != null)
             {
-                if (_connections == null)
-                {
-                    _connections = new Hashtable();
-                }
+                req.Credentials = credentials;
+            }
+            if (proxy != null)
+            {
+                req.Proxy = proxy;
+            }
+            if (cachePolicy != null)
+            {
+                req.CachePolicy = cachePolicy;
             }
 
-            HttpResponseMessage resp = await client.SendAsync(req).ConfigureAwait(false);
-
-            Stream respStream = new MemoryStream();
-            await resp.Content.CopyToAsync(respStream);
-            return respStream;
+            WebResponse resp = await Task<WebResponse>.Factory.FromAsync(req.BeginGetResponse, req.EndGetResponse, null).ConfigureAwait(false);
+            return resp.GetResponseStream();
```

Comment:
I believe this is being used by XmlUrlResolver which gives ownership of the stream to the caller - Does WebResponse live until it is disposed or can connection timeout and dispose of the stream or parts of data? @sepidehMS I believe you should check if that can be the case (IMO very likely) and cache the stream as it was implemented previously